share/doc/wake/basic_wake_tutorial.md

```diff
@@ -0,0 +1,660 @@
+## Wake Tutorial
+
+### Invoking wake
+
+1. Create a wake database: `wake --init .`
+
+2. Run command for wake: `wake -vx '<function_name> <parameter_if_any>'`
+
+3. To know about various wake options run `wake --help`. `v` stands for verbosity and `x` stands for execute the expression.
+
+4. Create a file with `.wake` extension and copy the functions mentioned in this tutorial and run them using `wake` command in the terminal.  
+
+5. For more information on wake library click [here](https://sifive.github.io/wake/)
+
+### wake `Hello World`
+* If you want to display a string you can do that using the below function by writing the string to be displayed inside "" and call the function abc . Ex: `wake -x 'abc'`. This will call the function `abc` and prints `Hello World`. 
+```
+def abc = "Hello World"
+```
+
+* The other way to do this is `wake -x '"Hello World"'`
+
+### Defining Functions
+A function may be defined using the keyword `def`. Simple function definition takes the form
+`def <function name> <parameter list> = <expression>`
+
+* Below function  double takes an argment x and gives the output by multiplying the x by 2
+```
+def double x = 2*x
+```
+
+* Below function triple takes an argument x and gives the output by multiplying the x by 3
+```
+def triple x = 3*x
+```
+
+* Function inc increments the values by 1 and returns the value
+```
+def inc x = x+1
+```
+
+* Function adda takes an argument of type string and concatenates it with b and gives the output. `{ }` is the string interpolation operator
```

Comment:
I'm not really sure what I am learning as a wake learner of all these different examples of functions. Can you clarify what you expect to get from each one? Like, I think the string interpolator example is good in that it shows the string interpolation operator, but that is probably more useful in a heading like "operations on strings" vs "functions". And the one below double (double x) is showing what, the use of parenthesis? Calling functions within another function? What should I take from that example?